hydromt_wflow/wflow.py

```diff
@@ -2715,6 +2716,267 @@ def setup_precip_forcing(
+scipy.interpolate. 2) “barnes”, from metpy.interpolate.
+
+        precip_stations_fn : str, gpd.GeoDataFrame
+            Source for the locations of the stations as points: (x, y) or (lat, lon).
+
+        index_col : str, optional
+            Column in precip_stations_fn to use for station ID values, by default None.
+
+        buffer: int, optional
+            Number of cells to use around the basins as a buffer to determine \
+which stations to include. Set to 100 cells by default.
+
+        metpy_kwargs: dict, optional
+            Keyword arguments to pass to `metpy.interpolate.interpolate_to_grid`.
+
+        **kwargs
+            Additional keyword arguments passed to the MetPy interpolation function. \
+See also ::py:meth:`workflows.forcing.spatial_interpolation`.
+
+        See Also
+        --------
+        hydromt_wflow.workflows.forcing.spatial_interpolation
+        metpy.interpolate.interpolate_to_grid
+        """
+        self.logger.warning("""Use of setup_precip_from_timeseries \
+is still in development.""")
+        # TODO remove logging message above when publishing
+        metpy_kwargs = metpy_kwargs or {}
+
+        starttime = self.get_config("starttime")
+        endtime = self.get_config("endtime")
+        freq = pd.to_timedelta(self.get_config("timestepsecs"), unit="s")
+        mask = self.grid[self._MAPS["basins"]].values > 0
+
+        # Use buffer around basins if provided, else limit to stations covered by basin
+        geom_buffered = (
+            self.basins.buffer(buffer * max(np.abs(self.res)))
+            if buffer
+            else self.basins
+        )
+
+        # Use basin centroid as 'station' for uniform case
+        if interp_type == "uniform":
+            precip_stations_fn = gpd.GeoDataFrame(
+                data=None,
+                geometry=[self.basins.unary_union.centroid],
+                index=["_station"],
+                crs=self.crs,
+            )
+
+        # Load the stations and their coordinates
+        if (
+            isinstance(precip_stations_fn, gpd.GeoDataFrame)
+            or isfile(precip_stations_fn)
+            or precip_stations_fn in self.data_catalog
+        ):
+            gdf_stations = self.data_catalog.get_geodataframe(
+                precip_stations_fn,
+                geom=geom_buffered,
+                assert_gtype="Point",
+                handle_nodata=NoDataStrategy.IGNORE,
+            )
+            # Use station ids from gdf_stations when reading the DataFrame
+            station_ids = (
+                gdf_stations[index_col] if index_col else gdf_stations.index
+            )
+
+            gdf_inside = gdf_stations.to_crs(self.crs).within(self.basins.unary_union)
+            self.logger.info(
+                f"""Found {len(station_ids)} stations in {precip_stations_fn} \
+(of which {gdf_inside.sum()} are located inside the basin), using a buffer \
+of {buffer} cells."""
```

Comment:
This info can be moved to interp function. This will avoid repeat if we want to implement for temperature/pet or when we do support geodataset.